src/components/menu/menu.ts

```diff
@@ -105,23 +120,48 @@ export class Menu extends Component<Props, SpreadsheetChildEnv> {
   static defaultProps = {
     depth: 1,
   };
+  private clickableMenuItems: Action[] = [];
+  // @ts-ignore: childrenHaveIcon will be used in XML
+  private childrenHaveIcon: boolean = false;
   private subMenu: MenuState = useState({
     isOpen: false,
     position: null,
     scrollOffset: 0,
     menuItems: [],
+    menuIndex: this.props.shouldSelectFirstItem ? 0 : undefined,
+    shouldSelectFirstItem: false,
   });
   private menuRef = useRef("menu");
   private position: DOMCoordinates = useAbsoluteBoundingRect(this.menuRef);
+  private menuItemRefs: { action: Action; ref: Ref<HTMLElement> }[] = [];
 
   setup() {
+    this.childrenHaveIcon = this.props.menuItems.some(
+      (menuItem) => !!menuItem.icon || !!menuItem.isActive
+    );
+    this.clickableMenuItems = this.props.menuItems.filter(
+      (menuItem) => menuItem.isEnabled(this.env) && menuItem.isVisible(this.env)
+    );
```

Comment:
@hokolomopo You're right, so I add a hook to calculate `clickableMenuItems` and `childrenHaveIcon`. 
For refs, since I need to use `useRef`, I cannot update them in the render hook, so I make it change with `menuItems` in props. I also add key to each menu to force rendering when switching different menus.